homeassistant/components/tesla_fleet/number.py

```diff
@@ -64,7 +64,7 @@ class TeslaFleetNumberVehicleEntityDescription(NumberEntityDescription):
         mode=NumberMode.AUTO,
         min_key="charge_state_charge_limit_soc_min",
         max_key="charge_state_charge_limit_soc_max",
-        func=lambda api, value: api.set_charge_limit(value),
+        func=lambda api, value: api.set_charge_limit(int(value)),
```

Comment:
You only pass on `int` here. Why not change the callable signature to be `Callable[[..], int]` instead? This way you don't have to explicitly do `int` everywhere (unless you expect something in the future I don't)

Reply:
the lambda can guarnatee it by just making the lambda get the value as an int and move the `int` to the number entity

Reply:
Changed the description to require an int.